katharsis-core/src/main/java/io/katharsis/resource/ResourceIdentifier.java

```diff
@@ -0,0 +1,79 @@
+package io.katharsis.resource;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class ResourceIdentifier implements Comparable<ResourceIdentifier> {
+
+	private String id;
+
+	private String type;
+
+	public ResourceIdentifier() {
+	}
+
+	public ResourceIdentifier(String id, String type) {
+		this.id = id;
+		this.type = type;
+	}
+
+	public String getId() {
+		return id;
+	}
+
+	public void setId(String id) {
+		this.id = id;
+	}
+
+	public String getType() {
+		return type;
+	}
+
+	public void setType(String type) {
+		this.type = type;
+	}
+
+	@Override
+	public ResourceIdentifier clone() {
+		return new ResourceIdentifier(id, type);
+	}
+
+	public static Object fromData(Object data) {
+		if (data == null) {
+			return null;
+		}
+		if (data instanceof Iterable) {
+			List<ResourceIdentifier> result = new ArrayList<>();
+			for (ResourceIdentifier id : (Iterable<ResourceIdentifier>) data) {
+				result.add(id.clone());
+			}
+			return result;
+		} else {
+			ResourceIdentifier id = (ResourceIdentifier) data;
+			return id.clone();
+		}
+	}
+
+	@Override
+	public int hashCode() {
+		return Objects.hash(id, type);
+	}
+
+	@Override
+	public boolean equals(Object obj) {
```

Comment:
manual hash/equals should only be required in rare cases anymore. This class is a lot of boilerplate